src/views-components/context-menu/actions/file-viewer-actions.tsx

```diff
@@ -0,0 +1,81 @@
+// Copyright (C) The Arvados Authors. All rights reserved.
+//
+// SPDX-License-Identifier: AGPL-3.0
+
+import * as React from "react";
+import { ListItemText, ListItem, ListItemIcon, Icon } from "@material-ui/core";
+import { RootState } from '~/store/store';
+import { getNodeValue } from '~/models/tree';
+import { CollectionDirectory, CollectionFile, CollectionFileType } from '~/models/collection-file';
+import { FileViewerList, FileViewer } from '~/models/file-viewers-config';
+import { getFileViewers } from '~/store/file-viewers/file-viewers-selectors';
+import { connect } from 'react-redux';
+import { OpenIcon } from '~/components/icon/icon';
+
+interface FileViewerActionProps {
+    fileUrl: string;
+    viewers: FileViewerList;
+}
+
+const mapStateToProps = (state: RootState): FileViewerActionProps => {
+    const { resource } = state.contextMenu;
+    if (resource) {
```

Comment:
Maybe better in this way? 
```
const file = resource ? getNodeValue(resource.uuid)(state.collectionPanelFiles) :  null;
const fileViewers = file ? getFileViewers(state.properties) : null;
return {
  fileUrl: file ? file.url : '',
  viewers: fileViewers ? fileViewers.filter(enabledViewers(file)): []
}
````

Reply:
In my opinion there's no major advantage in changing the current form.